keps/sig-node/20190505-pod-level-resource-quota.md

```diff
@@ -0,0 +1,144 @@
+---
+title: Pod Level Resource Quota
+authors:
+  - "@resouer"
+  - "@qinguoan"
+  - "@zhangxiaoyu-zidif"
+owning-sig: sig-node
+participating-sigs:
+  - sig-node
+reviewers:
+  - "@DawnChen"
+  - "@yujuhong"
+approvers:
+  - "@DawnChen"
+  - "@yujuhong"
+editor: TBD
+creation-date: 2019-05-05
+last-updated: 2019-05-05
+status: provisional
+---
+
+# Table of Contents
+
+- [Table of Contents](#table-of-contents)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+  - [Implementation Details](#implementation-details)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Implementation History](#implementation-history)
+- 
+
+## Summary
+The purpose of this enhancement is to redefine QoS class and Pod resource capacity's measurement methods.
```

Comment:
Very possible I'm missing something, but I'm not totally following how this summary relates to the title of "Add Pod Level Resource Quota".

Reply:
There is an example to express it:
If we have two containers in one pod, one container has resource limits(2c 4Gi, for example), the other container is sidecar with resource limit(1 c 1Gi), but we want the whole pod resource limit is 2c 4Gi, not the current(2+1 c and 4+1 Gi). That is saying, sidecar will share some resource from main container. And in pod level cgroup quota, the cgroups limit = sum(main container cgroups' limits)